api/api.go

```diff
@@ -37,6 +39,43 @@ type SubmissionInfo struct {
 	State string `json:"state"`
 }
 
+// StatusInfo contains information about a user's status on a particular language track.
+type StatusInfo struct {
+	Language string `json:"track_id"`
+	Recent   struct {
+		Problem   string `json:"problem"`
+		Submitted string `json:"submitted_at"`
+	}
+	Fetched []string `json:"fetched"`
+	Skipped []string `json:"skipped"`
+}
+
+func (s *StatusInfo) String() {
+	if len(s.Fetched) == 0 && s.Recent.Problem == "" {
+		fmt.Println("\nYou have yet to begin the", s.Language, "track!")
+	} else {
+		fmt.Println("\nYour status on the", s.Language, "track:")
+
+		fmt.Println("\nMost recently submitted exercise")
+
+		submittedAt, err := time.Parse(time.RFC3339Nano, s.Recent.Submitted)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Printf("%v (submitted on %v)\n", s.Recent.Problem, submittedAt)
```

Comment:
This is another good reason to make `Recent` its own type. I think give this and the input above, I'll go ahead and do that now.